### PR TITLE
Set macOS minimum version to macOS 11 Big Sur

### PR DESCRIPTION
**Pull request by shinra-electric**
I saw on Discord that the current minimum version for Citra is macOS 11, whereas it currently is set to 10.10 Yosemite (probably automatically). 

This PR adds the minimum version key to Info.plist with a MACOSX_MINIMUM_SYSTEM_VERSION variable, which it can be set in CMakeLists.txt. 

With this PR, the app can't be launched on 10.15 Catalina or lower, with a dialog explaining to the user and prompting them to update their OS. 

This should help to reduce the number of invalid issues reported on GitHub and Discord. 

Testing: 
Ensure the CI builds, and that the app bundle reports macOS 11 as the required version. 

Screenshot 2023-11-20 at 11 14 18

- - -
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/7196)

### CODE REVIEW

**Comment by Steveice10** [comment]
Wasn't aware we need to do more than just set the `CMAKE_OSX_DEPLOYMENT_TARGET`, thanks. Will take a look at the CI outputs.

File: src/citra_qt/CMakeLists.txt [hunk 1]
```diff
@@ -269,6 +269,7 @@ if (APPLE)
         MACOSX_BUNDLE_SHORT_VERSION_STRING "${BUILD_FULLNAME}"
         MACOSX_BUNDLE_LONG_VERSION_STRING "${BUILD_FULLNAME}"
         MACOSX_BUNDLE_ICON_FILE "citra.icns"
+        MACOSX_MINIMUM_SYSTEM_VERSION "11.0"
```
Comment (Steveice10) [inline src/citra_qt/CMakeLists.txt:272]: Could you set this to the value of `CMAKE_OSX_DEPLOYMENT_TARGET`? Helps reduce the number of places to update this in the future.

Comment (shinra-electric) [reply src/citra_qt/CMakeLists.txt]: Done

**Comment by shinra-electric** [comment]
Hmm, something is going wrong: 

Screenshot 2023-11-20 at 11 56 27

**Comment by Steveice10** [comment]
I think the issue is CMake only defines these constants for setting data into the Info.plist file: https://cmake.org/cmake/help/latest/prop_tgt/MACOSX_BUNDLE_INFO_PLIST.html

**Comment by Steveice10** [comment]
Supposedly according to [this post](https://stackoverflow.com/a/42358031), custom variables work if you set them as global variables instead of target properties?

**Comment by Steveice10** [comment]
It feels like a bit of a hack but this works for me:
```diff
diff --git a/dist/apple/Info.plist.in b/dist/apple/Info.plist.in
index 7eb5809257..563e0fa7e2 100644
--- a/dist/apple/Info.plist.in
+++ b/dist/apple/Info.plist.in
@@ -21,6 +21,8 @@
     <string>${MACOSX_BUNDLE_INFO_STRING}</string>
     <key>NSHumanReadableCopyright</key>
     <string>${MACOSX_BUNDLE_COPYRIGHT}</string>
+    <key>LSMinimumSystemVersion</key>
+    <string>${MACOSX_MINIMUM_SYSTEM_VERSION}</string>
     <!-- Fixed -->
     <key>LSApplicationCategoryType</key>
     <string>public.app-category.games</string>
diff --git a/src/citra_qt/CMakeLists.txt b/src/citra_qt/CMakeLists.txt
index b33729ffbc..a751cc0d3c 100644
--- a/src/citra_qt/CMakeLists.txt
+++ b/src/citra_qt/CMakeLists.txt
@@ -260,6 +260,9 @@ if (APPLE)
 
     # Define app bundle metadata.
     include(GenerateBuildInfo)
+    # This is not an official CMake configure property for Info.plist files.
+    # However, if we set it as a variable instead it seems to work to substitute in.
+    set(MACOSX_MINIMUM_SYSTEM_VERSION ${CMAKE_OSX_DEPLOYMENT_TARGET})
     set_target_properties(citra-qt PROPERTIES
         MACOSX_BUNDLE TRUE
         MACOSX_BUNDLE_INFO_PLIST "${DIST_DIR}/Info.plist.in"
```

**Comment by Steveice10** [comment]
Or alternatively, it can be as simple as this:
```diff
diff --git a/dist/apple/Info.plist.in b/dist/apple/Info.plist.in
index 7eb5809257..684c7e722f 100644
--- a/dist/apple/Info.plist.in
+++ b/dist/apple/Info.plist.in
@@ -21,6 +21,8 @@
     <string>${MACOSX_BUNDLE_INFO_STRING}</string>
     <key>NSHumanReadableCopyright</key>
     <string>${MACOSX_BUNDLE_COPYRIGHT}</string>
+    <key>LSMinimumSystemVersion</key>
+    <string>${CMAKE_OSX_DEPLOYMENT_TARGET}</string>
     <!-- Fixed -->
     <key>LSApplicationCategoryType</key>
     <string>public.app-category.games</string>
```

**Comment by shinra-electric** [comment]
> Or alternatively, it can be as simple as this:

Yes, this makes more sense....

**Review by Steveice10 — APPROVED** [review]
Checked CI and looks good, thanks!

**Pull request merged**